lighthouse-core/lib/asset-saver.js

```diff
@@ -216,24 +225,24 @@ function* traceJsonGenerator(traceData) {
  * @param {string} traceFilename
  * @return {Promise<void>}
  */
-function saveTrace(traceData, traceFilename) {
-  return new Promise((resolve, reject) => {
-    const traceIter = traceJsonGenerator(traceData);
-    // A stream that pulls in the next traceJsonGenerator chunk as writeStream
-    // reads from it. Closes stream with null when iteration is complete.
-    const traceStream = new stream.Readable({
-      read() {
-        const next = traceIter.next();
-        this.push(next.done ? null : next.value);
-      },
-    });
+async function saveTrace(traceData, traceFilename) {
+  const traceIter = traceJsonGenerator(traceData);
+  const writeStream = fs.createWriteStream(traceFilename);
 
-    const writeStream = fs.createWriteStream(traceFilename);
-    writeStream.on('finish', resolve);
-    writeStream.on('error', reject);
+  return pipeline(stream.Readable.from(traceIter), writeStream);
```

Comment:
this was able to get way nicer than the old version. Node 13+ will let us drop the `stream.Readable.from()` part (`pipeline` will take iterables directly) and node 15+ will let us drop the `promisify(stream.pipeline)` call

Reply:
add a compat TODO comment for the latter bit?

Reply:
yah